src/plugins/chart_expressions/expression_partition_vis/public/components/partition_vis_component.tsx

```diff
@@ -213,49 +233,53 @@ const PieComponent = (props: PieComponentProps) => {
     },
   });
 
-  const { bucketColumns, metricColumn } = useMemo(
-    () => getColumns(visParams, visData),
-    [visData, visParams]
-  );
-
+  const isDarkMode = props.chartsThemeService.useDarkMode();
   const layers = useMemo(
     () =>
       getLayers(
+        visType,
         bucketColumns,
         visParams,
+        visData,
         props.uiState?.get('vis.colors', {}),
         visData.rows,
         props.palettesRegistry,
         services.fieldFormats,
-        syncColors
+        syncColors,
+        isDarkMode
       ),
     [
+      visType,
       bucketColumns,
       visParams,
+      visData,
       props.uiState,
       props.palettesRegistry,
-      visData.rows,
       services.fieldFormats,
       syncColors,
+      isDarkMode,
     ]
   );
 
   const rescaleFactor = useMemo(() => {
     const overallSum = visData.rows.reduce((sum, row) => sum + row[metricColumn.id], 0);
     const slices = visData.rows.map((row) => row[metricColumn.id] / overallSum);
-    const smallSlices = slices.filter((value) => value < 0.02).length;
-    if (smallSlices) {
+    const smallSlices = slices.filter((value) => value < 0.02) ?? [];
+    if (smallSlices.length) {
       // shrink up to 20% to give some room for the linked values
-      return 1 / (1 + Math.min(smallSlices * 0.05, 0.2));
+      return 1 / (1 + Math.min(smallSlices.length * 0.05, 0.2));
     }
     return 1;
   }, [visData.rows, metricColumn]);
 
   const config = useMemo(
-    () => getConfig(visParams, chartTheme, dimensions, rescaleFactor),
-    [chartTheme, visParams, dimensions, rescaleFactor]
+    () => getConfig(visType, visParams, chartTheme, dimensions, rescaleFactor),
+    [visType, visParams, chartTheme, dimensions, rescaleFactor]
   );
+
+  const fixedViewPort = document.getElementById('app-fixed-viewport');
```

Comment:
Why are we doing this?

Reply:
It's an elastic-charts feature to prevent the tooltip from overlaying the Kibana chrome - if the boundary is specified the tooltip will always stay within the app. This is especially important for large tooltips. We are doing the same in Lens: https://github.com/elastic/kibana/blob/b0442e396b360f788524cfe38de62fa31fc1d789/x-pack/plugins/lens/public/xy_visualization/expression.tsx#L648

Reply:
Got it! Thanx Joe :)